### DISCUSSION TITLE
Oil Temp Always Showing 32 degrees F

### DISCUSSION THREAD

**Discussion by ruckusvol** [BimmerGestalt/AAIdrive#571] [Q&A]
Everything else in the car info works but my oil always shows 32 degrees F (or 0 Celsius).  Is it working for others?  Anything I can do to help diagnose it?

**Comment by hufman** [comment]
The native car unit for this temperature point is Celcius, which means your car is reporting 0 for this data point. My car also doesn't emit valid data for this data point, but I assumed it was because it's electric. I'd be interested to hear what other people say!

**Reply by ruckusvol** [reply]
Is there a way for me to access all these variables manually?  I have E-sys, OBD2, DME logging, etc available.  Just don’t know where to look.  If I do I can check on what’s there and what’s working.  I know DME logging properly shows oil temp.

**Reply by hufman** [reply]
The CDS system is relatively high-level, responding with JSON over the Apache Etch RPC connection, so I'm not sure where the data comes from internally. You can install the [CDS Details](https://github.com/BimmerGestalt/IDriveConnectAddons/releases/download/latest_build/cds_details-debug.apk) Addon to view every CDS data point that your car provides, maybe that will help provide clues!

**Reply by ruckusvol** [reply]
Oh cool!  Just installed it and I’ll play with it next time I’m in the car.  Are you open to ideas/suggestions if I find anything that might be helpful?

**Reply by hufman** [reply]
Sure, let me know what you find!

**Reply by ruckusvol** [reply]
I did find that the oil temp is working correctly in the CDS Details.  But the last reading when I leave the car is 0.  I wonder if it starts reporting 0 when the engine stops running.  I'll check it next time.

**Reply by dinan5m3** [reply]
Hi, just getting on this project, starting with CDS Gauges and CDS Details addon's.  Is there any provision to display CDS Gauges information on the car's IDrive display?  Seeing if with some code changes, the Engine Oil Temp value can be displayed in the IDrive screen?  Thanks!

**Reply by hufman** [reply]
There is no support for showing any gauges in the car. I would need to find an existing Android BMW App to log in to the car, and the only matching app I know about is the Mini Connected app for iOS with its Sports Displays app. The iOS app certs don't work on Android however.

**Reply by dinan5m3** [reply]
Thanks.  What do you mean by, "I would need to find an existing Android BMW App to log into the car"?  

There is already an existing Android BMW App to log into the car, right?

**Reply by hufman** [reply]
A phone app requires official BMW car resources to log in to the car, which is why AAIdrive shows up in the car as Spotify and SmartThings. I haven't found any BMW-compatible app for Android that provides a gauge display in its car resources, which prevents any other app from implementing a gauge display.

**Reply by dinan5m3** [reply]
What you are saying is, in Mini cars there is official car resource called Sports Display (i.e. Sport Display app, similar to BMW Spotify app in IDRIVE) that the Mini Connected app for iOS can log into, and provide gauge display in Mini car resources?  

Do you have any picture of this working in Mini where iOS app implements gauge display in Mini? 

Thanks.

**Reply by dinan5m3** [reply]
I see what you are saying, you are referring to an "old" Mini Sports Displays iOS app.

Just curious, how long ago was this Mini Sports Displays iOS app available?

This old app was able to implement a gauge display because it had matching car resource in Mini?
